### PR TITLE
Added Interface and Classes for Stategy Design Pattern for Filtering

### PR DESCRIPTION
**Pull request by rickypramanick**
As per #172

### CODE REVIEW

**Comment by awesominat** [comment]
I don't think these should be in the use case. here's the step-by-step process I envision for this feature

**Keeping it in the use case: you'd have to**
1. Changing input data interface and call parameters to include the filter TYPE (this will have to be a string)
1.5 (violation of Liskov's rule). You will have to create switch statements to handle which strategy should be instantiated based on the filter type
2. Changing use case to filter the entire list before sending it (AFTER unnecessarily recomputing all the transactions, could lag if there are a lot of transactions and is a lot of unnecessary calculations)
3. Change how the presenter functions to update the state
4. Leave view unchanged

or
**Putting the code in the interface_adapters**
1. add a strategy variable to view. If the user clicks one of the filters, instantiate a new filter almost instantly
2. call strategy.filter(old list passed into the table model) (this old list is the list you were displaying previous)

tl:dr, it is much better to put these files in the interface_adapters because
1. doesn't violate Liskov's
2. doesn't unnecessary increase complexity
3. easier to read

**Review by gursi26 — CHANGES_REQUESTED** [review]
Tested locally and I have noticed a couple of bugs
- The default selection when opening transaction history view is null rather than No filter. This means my table is empty and the table only shows up when I manually select No filter. Please make it so that the default selection when navigating to this page is No filter instead.
- When attempting to select options in both filter fields (for example, BUY transactions for Apple stock), the other field switches to null. This means that it is not possible to apply both fields simulataneously.
- Restructure the screen so that the dropdown menus appear one on top of the other rather than side-by-side, because they take up too much space in their current orientation. Also move the back button to the top left of the screen.

**Review by gursi26 — APPROVED** [review]
